app/grandchallenge/statistics/views.py

```diff
@@ -41,135 +45,295 @@ def get_context_data(self, **kwargs):
         days = 30
         max_num_results = 10
 
-        User = get_user_model()  # noqa: N806
-
         time_period = timezone.now() - timedelta(days=days)
 
         public_challenges = Challenge.objects.filter(hidden=False)
 
-        country_data = (
-            User.objects.exclude(user_profile__country="")
-            .values("user_profile__country")
-            .annotate(country_count=Count("user_profile__country"))
-            .order_by("-country_count")
-            .values_list("user_profile__country", "country_count")
-        )
+        stats = cache.get(settings.STATISTICS_SITE_CACHE_KEY)
 
-        extra = {
-            "days": days,
-            "max_num_results": max_num_results,
-            "number_of_users": User.objects.filter(is_active=True).count(),
-            "country_data": [
-                {
-                    "id": countries.numeric(c[0], padded=True),
-                    "participants": c[1],
-                }
-                for c in country_data
-            ],
-            "new_users_period": (
-                User.objects.filter(date_joined__gt=time_period).count()
-            ),
-            "logged_in_period": (
-                User.objects.filter(last_login__gt=time_period).count()
-            ),
-            "public_challenges": public_challenges.count(),
-            "hidden_challenges": Challenge.objects.filter(hidden=True).count(),
-            "submissions": Submission.objects.count(),
-            "submissions_period": (
-                Submission.objects.filter(created__gt=time_period).count()
-            ),
-            "latest_public_challenge": (
-                public_challenges.order_by("-created").first()
-            ),
-            "mp_group": (
-                Group.objects.filter(
-                    participants_of_challenge__in=public_challenges
-                )
-                .annotate(num_users=Count("user"))
-                .order_by("-num_users")
-                .first()
-            ),
-            "challenge_registrations_period": (
-                self._challenge_qs_to_list_with_url(
-                    public_challenges.filter(
-                        registrationrequest__created__gt=time_period
+        if stats is None:
+            update_site_statistics_cache()
+            stats = cache.get(settings.STATISTICS_SITE_CACHE_KEY)
```

Comment:
This is only necessary once, in the first couple of hours after deployment until the task has been run for the first time, right? Wouldn't it be easier to run it in the shell after deploying then?

Reply:
Ok, thanks for explaining.